### PR TITLE
KAFKA-7253 The connector type responded by worker is always null when…

### PR DESCRIPTION
**Pull request by chia7712**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
ReassignPartitionsClusterTest pass locally

**Comment by chia7712** [comment]
There already is a related test - DistributedHerderTest#testCreateConnector. However, ConnectorInfo#equals doesn't compare the connector type so DistributedHerderTest#testCreateConnector doesn't discover the bug.

**Comment by chia7712** [comment]
@hachikuji Could you please take a look if you have free cycle? thanks!

**Comment by chia7712** [comment]
Ping for reviews since this bug is related to public APIs of connector

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorInfo.java [hunk 1]
```diff
@@ -71,7 +71,8 @@ public boolean equals(Object o) {
         ConnectorInfo that = (ConnectorInfo) o;
         return Objects.equals(name, that.name) &&
                 Objects.equals(config, that.config) &&
-                Objects.equals(tasks, that.tasks);
+                Objects.equals(tasks, that.tasks) &&
+                Objects.equals(type, that.type);
```
Comment (hachikuji) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorInfo.java:75]: We should update `hashCode` as well?

**Review by hachikuji — APPROVED** [review]
Left a minor comment, but otherwise LGTM.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorInfo.java (hunk 1 above)
Comment (chia7712) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorInfo.java:75]: @hachikuji You are right. Will update the PR asap

**Comment by hachikuji** [comment]
The failures appear to be unrelated. I will go ahead and merge. Thanks for the patch!

**Comment by chia7712** [comment]
> The failures appear to be unrelated. I will go ahead and merge. Thanks for the patch!

Thanks for the review!

**Comment by chia7712** [comment]
@hachikuji Is this PR backported to 1.x? This PR fix a bug about public APIs (I assume connector restful APIs is a part of kafka's public APIs)

**Pull request merged**